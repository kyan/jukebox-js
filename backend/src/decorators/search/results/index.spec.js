import SearchResults from './index'
import fs from 'fs'
import lolex from 'lolex'

describe('SearchResults', () => {
  let clock
  const payload = JSON.parse(fs.readFileSync('./src/__mockData__/searchResults.json', 'utf8'))

  beforeEach(() => {
    clock = lolex.install()
  })

  afterEach(() => {
    clock = clock.uninstall()
  })

  describe('when passed a tracklist of varying data', () => {
    it('transforms it', () => {
      expect(SearchResults(payload.tracks.items)).toEqual([{ 'track': { 'album': { 'name': 'Ken Dodd', 'uri': 'spotify:album:1D7E2EHMoQM3nP9drEwv7o', 'year': undefined }, 'artist': { 'name': 'Ken Dodd', 'uri': 'spotify:artist:76o4kCpWMmBGl8jIYfRHTk' }, 'image': 'https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92', 'length': 120066, 'name': 'Happiness', 'uri': 'spotify:track:1WaBuuaXGwrU4sFvxAjnkf' } }, { 'track': { 'album': { 'name': 'Ken Dodd - His Greatest Hits', 'uri': 'spotify:album:59TyQORxcvy9RWj7gkZMvB', 'year': undefined }, 'artist': { 'name': 'Ken Dodd', 'uri': 'spotify:artist:76o4kCpWMmBGl8jIYfRHTk' }, 'explicit': true, 'image': 'https://i.scdn.co/image/ab67616d0000b27368b6e0998ac2c1726839bdcd', 'length': 120066, 'name': 'Happiness', 'uri': 'spotify:track:6idaUJ1KK1mWyxQziMefhU' } }, { 'track': { 'album': { 'name': 'Ken Dodd', 'uri': 'spotify:album:1D7E2EHMoQM3nP9drEwv7o', 'year': undefined }, 'artist': { 'name': 'Ken Dodd', 'uri': 'spotify:artist:76o4kCpWMmBGl8jIYfRHTk' }, 'image': 'https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92', 'length': 163026, 'name': 'Love Me with All Your Heart (Cuando Calienta El Sol)', 'uri': 'spotify:track:7bqfZygxuq03hqffjswjCK' } }])
    })
  })
})
