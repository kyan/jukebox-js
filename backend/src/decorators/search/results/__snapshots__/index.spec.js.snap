// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults when passed no results renders 1`] = `
Array [
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "123",
            "fullname": "Big Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 120066,
      "name": "Happiness",
      "uri": "spotify:track:1WaBuuaXGwrU4sFvxAjnkf",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd - His Greatest Hits",
        "uri": "spotify:album:59TyQORxcvy9RWj7gkZMvB",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b27368b6e0998ac2c1726839bdcd",
      "length": 120066,
      "name": "Happiness",
      "uri": "spotify:track:6idaUJ1KK1mWyxQziMefhU",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 163026,
      "name": "Love Me with All Your Heart (Cuando Calienta El Sol)",
      "uri": "spotify:track:7bqfZygxuq03hqffjswjCK",
    },
  },
]
`;

exports[`SearchResults when passed results with no votes renders 1`] = `
Array [
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "123",
            "fullname": "Big Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 120066,
      "name": "Happiness",
      "uri": "spotify:track:1WaBuuaXGwrU4sFvxAjnkf",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd - His Greatest Hits",
        "uri": "spotify:album:59TyQORxcvy9RWj7gkZMvB",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "explicit": true,
      "image": "https://i.scdn.co/image/ab67616d0000b27368b6e0998ac2c1726839bdcd",
      "length": 120066,
      "name": "Happiness",
      "uri": "spotify:track:6idaUJ1KK1mWyxQziMefhU",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 163026,
      "name": "Love Me with All Your Heart (Cuando Calienta El Sol)",
      "uri": "spotify:track:7bqfZygxuq03hqffjswjCK",
    },
  },
]
`;

exports[`SearchResults when passed search results of votes renders 1`] = `
Array [
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 163026,
      "metrics": Object {
        "votesAverage": 50,
      },
      "name": "Love Me with All Your Heart (Cuando Calienta El Sol)",
      "uri": "spotify:track:7bqfZygxuq03hqffjswjCK",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd - His Greatest Hits",
        "uri": "spotify:album:59TyQORxcvy9RWj7gkZMvB",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b27368b6e0998ac2c1726839bdcd",
      "length": 120066,
      "metrics": Object {
        "votesAverage": 30,
      },
      "name": "Happiness",
      "uri": "spotify:track:6idaUJ1KK1mWyxQziMefhU",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "123",
            "fullname": "Big Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 120066,
      "name": "Happiness",
      "uri": "spotify:track:1WaBuuaXGwrU4sFvxAjnkf",
    },
  },
]
`;

exports[`SearchResults when passed search results with deincreasing votes renders 1`] = `
Array [
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "123",
            "fullname": "Big Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 120066,
      "metrics": Object {
        "votesAverage": 80,
      },
      "name": "Happiness",
      "uri": "spotify:track:1WaBuuaXGwrU4sFvxAjnkf",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd - His Greatest Hits",
        "uri": "spotify:album:59TyQORxcvy9RWj7gkZMvB",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b27368b6e0998ac2c1726839bdcd",
      "length": 120066,
      "metrics": Object {
        "votesAverage": 50,
      },
      "name": "Happiness",
      "uri": "spotify:track:6idaUJ1KK1mWyxQziMefhU",
    },
  },
  Object {
    "track": Object {
      "addedBy": Array [
        Object {
          "user": Object {
            "_id": "456",
            "fullname": "Bigger Rainbowhead",
          },
        },
      ],
      "album": Object {
        "name": "Ken Dodd",
        "uri": "spotify:album:1D7E2EHMoQM3nP9drEwv7o",
        "year": undefined,
      },
      "artist": Object {
        "name": "Ken Dodd",
        "uri": "spotify:artist:76o4kCpWMmBGl8jIYfRHTk",
      },
      "image": "https://i.scdn.co/image/ab67616d0000b2731ea0a21d873289c511dfaf92",
      "length": 163026,
      "metrics": Object {
        "votesAverage": 30,
      },
      "name": "Love Me with All Your Heart (Cuando Calienta El Sol)",
      "uri": "spotify:track:7bqfZygxuq03hqffjswjCK",
    },
  },
]
`;
